File: MEMO.DAL/Entities/Translation.cs
using MEMO.DAL.Interfaces;
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Text;

namespace MEMO.DAL.Entities
{
    public class Translation : IAuditable
    {
        public Guid Id { get; set; }
        [MaxLength(30)]
        public string Original { get; set; }
        [MaxLength(30)]
        public string Translated { get; set; }
        public int Color { get; set; }

        public Guid DictionaryId { get; set; }
        public Dictionary Dictionary { get; set; }
        public DateTime CreationDate { get; set; }

        [NotMapped]
        public int AttributeValueCount { get; set; }

        public ICollection<AttributeValue> AttributeValues { get; set; } = new List<AttributeValue>();
    }
}

File: MEMO.MobilClient/app/src/main/java/hu/mobilclient/memo/model/memoapi/Translation.kt
package hu.mobilclient.memo.model.memoapi

import android.graphics.Color
import androidx.core.content.ContextCompat
import com.google.gson.annotations.SerializedName
import hu.mobilclient.memo.App
import hu.mobilclient.memo.R
import hu.mobilclient.memo.helpers.Constants
import hu.mobilclient.memo.model.memoapi.interfaces.IAuditable
import java.util.*

class Translation(@SerializedName("id")var Id: UUID = UUID(0,0),
                  @SerializedName("original")var Original: String = Constants.EMPTY_STRING,
                  @SerializedName("translated")var Translated: String = Constants.EMPTY_STRING,
                  @SerializedName("dictionaryId") var DictionaryId: UUID? = null,
                  @SerializedName("attributeValues") var AttributeValues: MutableList<AttributeValue> = ArrayList(),
                  @SerializedName("attributeValueCount")var AttributeValueCount: Int = Constants.ZERO,
                  @SerializedName("color")var Color: Int = ContextCompat.getColor(App.instance, R.color.translation_default),
                  @SerializedName("creationDate") override var CreationDate: String = Constants.EMPTY_STRING) : IAuditable